WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/ReaderViewPage.java

```diff
@@ -0,0 +1,106 @@
+package org.wordpress.android.e2e.pages;
+
+import android.view.KeyEvent;
+
+import androidx.test.platform.app.InstrumentationRegistry;
+import androidx.test.uiautomator.UiDevice;
+import androidx.test.uiautomator.UiObject;
+import androidx.test.uiautomator.UiSelector;
+
+import static junit.framework.TestCase.assertFalse;
+import static junit.framework.TestCase.assertTrue;
+import static org.wordpress.android.support.WPSupportUtils.DEFAULT_TIMEOUT;
+import static org.wordpress.android.support.WPSupportUtils.isTextDisplayed;
+import static org.wordpress.android.support.WPSupportUtils.swipeToLeft;
+import static org.wordpress.android.support.WPSupportUtils.swipeToRight;
+
+public class ReaderViewPage {
+    UiDevice mDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation());
+
+    String mLikerContainerId = "org.wordpress.android.prealpha:id/liker_faces_container";
+    String mRelatedPostsId = "org.wordpress.android.prealpha:id/container_related_posts";
+
+    UiObject mLikerContainer = mDevice.findObject(new UiSelector().resourceId(mLikerContainerId));
+    UiObject mRelatedPostsContainer = mDevice.findObject(new UiSelector().resourceId(mRelatedPostsId));
+    UiObject mSwipeForMore = mDevice.findObject(new UiSelector().textContains("Swipe for more"));
+
+    public ReaderViewPage waitUntilLoaded() {
+        mRelatedPostsContainer.waitForExists(DEFAULT_TIMEOUT);
+
+        return this;
+    }
+
+    public ReaderViewPage like() {
+        tapLikeButton();
+        mLikerContainer.waitForExists(DEFAULT_TIMEOUT);
+
+        return this;
+    }
+
+    public ReaderViewPage unlike() {
+        tapLikeButton();
+        mLikerContainer.waitUntilGone(DEFAULT_TIMEOUT);
+
+        return this;
+    }
+
+    private void tapLikeButton() {
+        mSwipeForMore.waitUntilGone(DEFAULT_TIMEOUT);
+        // Even though it was working locally in simulator, tapping the footer buttons,
+        // like 'mLikeButton.click()', was not working in CI.
+        // The current workaround is to use arrows navigation.
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_DOWN);
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_RIGHT);
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_RIGHT);
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_RIGHT);
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_CENTER);
+        resetFocus();
+    }
+
+    private void resetFocus() {
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_LEFT);
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_LEFT);
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_LEFT);
```

Comment:
Just curious as to why this is needed to press the same button 3 times?

Reply:
Yeah... Another "don't let my mother know" piece of code. =/ 

It was needed to reset the focus so we could `Like` and `Unlike` in sequence. To that, It was only needed to tap anywhere in the screen like the post title. The issue is that I couldn't find a reliable element to tap, the post title, for example, wasn't always found. I also tried interacting with the notification bar, which worked fine locally but not in CI.

Going back to your question, the `resetFocus()` is navigating back to the first button in the footer so we can call `tapLikeButton()` multiple times in sequence and always hit the `Like` button.

Reply:
I'm gonna change this because `resetFocus()` as it is right now will only work for the `Like` button, if we wanted to tap the Comments button it would only take 2 `KEYCODE_DPAD_LEFT`, so it's not reusable.